src/Interactive/EditorFeatures/Core/Extensibility/Interactive/NuGetPackageResolverImpl.cs

```diff
@@ -0,0 +1,180 @@
+
+                    // Read the references from project.lock.json.
+                    var projectLockJson = PathUtilities.CombineAbsoluteAndRelativePaths(tempPath, "project.lock.json");
+                    using (var stream = File.OpenRead(projectLockJson))
+                    using (var reader = new StreamReader(stream))
+                    {
+                        return ReadProjectLockJson(_packagesDirectory, reader);
+                    }
+                }
+                finally
+                {
+                    tempDir.Delete(recursive: true);
+                }
+            }
+            catch (IOException)
+            {
+            }
+            catch (UnauthorizedAccessException)
+            {
+            }
+            return default(ImmutableArray<string>);
+        }
+
+        /// <summary>
+        /// Syntax is "id/version", matching references in project.lock.json.
+        /// </summary>
+        internal static bool ParsePackageReference(string reference, out string name, out string version)
+        {
+            int separator = reference.IndexOf('/');
+            int n = reference.Length;
+            if ((separator > 0) &&
+                (separator < n - 1) &&
+                (reference.IndexOf('/', separator + 1) < 0))
+            {
+                name = reference.Substring(0, separator);
+                version = reference.Substring(separator + 1);
+                return true;
+            }
+            name = null;
+            version = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Generate a project.json file with the packages as "dependencies".
+        /// </summary>
+        internal static void WriteProjectJson(TextWriter writer, string packageName, string packageVersion)
+        {
+            using (var jsonWriter = new JsonTextWriter(writer))
+            {
+                jsonWriter.Formatting = Newtonsoft.Json.Formatting.Indented;
+                jsonWriter.WriteStartObject();
+                jsonWriter.WritePropertyName("dependencies");
+                jsonWriter.WriteStartObject();
+                jsonWriter.WritePropertyName(packageName, escape: true);
+                jsonWriter.WriteValue(packageVersion);
+                jsonWriter.WriteEndObject();
+                jsonWriter.WriteEndObject();
+            }
+        }
+
+        /// <summary>
+        /// Read the references from the project.lock.json file.
+        /// </summary>
+        internal static ImmutableArray<string> ReadProjectLockJson(string packagesDirectory, TextReader reader)
+        {
+            JObject obj;
+            using (var jsonReader = new JsonTextReader(reader))
+            {
+                obj = JObject.Load(jsonReader);
+            }
+            var builder = ArrayBuilder<string>.GetInstance();
+            var targets = (JObject)GetPropertyValue(obj, "targets");
+            // TODO: Choose the appropriate target. For now use the first.
```

Comment:
Once you specify a framework, then this will be stable. Honestly, hard-coding some arbitrary framework version is probably better than getting lucky like this...